File: packages/print-service-client/src/pos-cash-session-opening-ticket.ts
/**
 * Apertura de caja POS → agente KaiPrinters (`type: "pos-cash-session-opening-ticket"`).
 */

import type { PosSaleTicketCompany } from "./pos-sale-ticket";

export const POS_CASH_SESSION_OPENING_TICKET_PAYLOAD_VERSION = 1;

export type PosCashSessionOpeningTicketPayload = {
  version: number;
  cashSessionId: string;
  openedAt: string;
  openingAmount: number;
  company: PosSaleTicketCompany;
  branchName?: string | null;
  pointOfSaleName?: string | null;
  operatorName?: string | null;
  cashHubName?: string | null;
};

export type PosCashSessionOpeningTicketPrintExtras = {
  filename: string;
  documentType?: string;
  internalFolio?: string;
  sourceApp?: string;
  purpose?: string;
};

File: kai-printers-desktop/src-tauri/src/pos_cash_session_opening_ticket.rs
//! Apertura de caja POS → JSON (`pos-cash-session-opening-ticket`).

use crate::pos_sale_ticket_pdf::TicketCompany;
use anyhow::{Context, Result};
use serde::Deserialize;
use std::path::PathBuf;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosCashSessionOpeningTicket {
    #[serde(default)]
    #[allow(dead_code)]
    pub version: i32,
    pub cash_session_id: String,
    pub opened_at: String,
    pub opening_amount: f64,
    pub company: TicketCompany,
    pub branch_name: Option<String>,
    pub point_of_sale_name: Option<String>,
    pub operator_name: Option<String>,
    pub cash_hub_name: Option<String>,
}

pub fn parse_pos_cash_session_opening_ticket_from_value(
    value: &serde_json::Value,
) -> Result<PosCashSessionOpeningTicket> {
    serde_json::from_value(value.clone()).context("parse pos-cash-session-opening-ticket")
}

pub fn write_pos_cash_session_opening_ticket_escpos_from_value(
    dir: &PathBuf,
    value: &serde_json::Value,
) -> Result<PathBuf> {
    crate::pos_cash_session_opening_ticket_escpos::write_pos_cash_session_opening_ticket_escpos_from_value(
        dir, value,
    )
}
